### ISSUE TITLE
Refactor: improve error handling in GithubRepos::RepoSyncWorker

### ISSUE THREAD

**Issue by rhymes** [forem/forem#10106]
Currently we're inspecting the error message to determine if a user's account was suspended or if the repository has become unavailable so that we can destroy it

https://github.com/forem/forem/blob/25e20bdc1ce3e66282298a88cdf55f89b1b1e9a5/app/workers/github_repos/repo_sync_worker.rb#L32-L37

`Ockokit` (the GitHub library we use) has two specific errors for those two cases: `Octokit::RepositoryUnavailable` and `Octokit::AccountSuspended`.

It'd be nice to replace those messages inspection with the correct error classes.

See https://github.com/octokit/octokit.rb/blob/a69f5eab0a2fb2763fcda9fadcb0c961df96a373/lib/octokit/error.rb#L265 and https://github.com/octokit/octokit.rb/blob/a69f5eab0a2fb2763fcda9fadcb0c961df96a373/lib/octokit/error.rb#L257

By adding two exceptions similarly named, `RepositoryUnavailable` and `AccountSuspended`, to https://github.com/forem/forem/blob/master/app/services/github/errors.rb one should be able to handle them correctly in the worker, without inspecting the message and thus keeping the same behaviour

_Originally posted by @rhymes in https://github.com/forem/forem/pull/10103#discussion_r480245028_

**Comment by diogoosorio** [comment]
FYI I'll give this one a crack. 😄

**Comment by rhymes** [comment]
@diogoosorio are you still working on this? Thank you

**Comment by diogoosorio** [comment]
Hum, I kind of already did (??):

https://github.com/forem/forem/pull/10131/files#diff-1dca539dd1dfbbc75ce747cbd737698d

I haven't looked into it, it was either reverted for some reason or there was some sort of a concurrency issue which lead to git merge this incorrectly. I'll look into the version control history and publish my findings here. 😄

**Comment by diogoosorio** [comment]
Nevermind, this is in the state I was expecting on master:

https://github.com/forem/forem/blob/master/app/workers/github_repos/repo_sync_worker.rb#L30

Unless I entirely misunderstood the problem, I think this is done. If I have misunderstood something, please ELI5 it to me. 🙏

**Comment by rhymes** [comment]
You're right! Thanks @diogoosorio
